src/harness/sas_test_harness.py

```diff
@@ -0,0 +1,360 @@
+    self.server = SasHttpServer(self.dump_path,
+                                self.getUrlHostNameAndPort(),
+                                SasTestHarnessServerHandler)
+    self.server.socket = ssl.wrap_socket(
+        self.server.socket,
+        certfile=self.cert_file,
+        keyfile=self.key_file,
+        ca_certs=CA_CERT,
+        cert_reqs=ssl.CERT_REQUIRED,
+        ssl_version=ssl.PROTOCOL_TLSv1_2,
+        ciphers=':'.join(CIPHERS),
+        server_side=True
+    )
+
+    # Start HTTP Server.
+    logging.info("Started Test Harness Server:%s at %s", self.name, self.http_server_url)
+    self.server.serve_forever()
+
+  def shutdown(self):
+    self.server.shutdown()
+    logging.info("Stopped Test Harness Server:%s", self.name)
+
+  @staticmethod
+  def getCbsdReferenceId(fcc_id, serial_number):
+    """creates encoded cbsd_reference_id using sha1 with fcc_id and serial number of cbsd.
+    """
+    return str(fcc_id + '/' + str(hashlib.sha1(serial_number).hexdigest())).encode('utf-8')
+
+  @staticmethod
+  def writeDumpFile(base_path, file_name, data):
+    """ Write JSON data to specified file. """
+    dump_file_path = os.path.join(base_path, file_name)
+    file_handler = open(dump_file_path, "w")
+    file_handler.writelines(json.dumps(data, sort_keys=True, indent=4))
+    file_handler.close()
+
+  @staticmethod
+  def cleanDumpFiles(base_path):
+    """ Clean existing dumpfile if any. """
+    map(os.remove, (os.path.join(base_path, file_path)
+                    for file_path in os.listdir(base_path)
+                    if os.path.exists(os.path.join(base_path, file_path))))
+
+  @staticmethod
+  def createFadRecord(url, sas_version, record):
+    """ Creates the files object that includes url,checksum,size,version and recordtype of record
+        required to be packed in FAD.
+    """
+    fad_record = {}
+    fad_record['url'] = url
+    fad_record['checksum'] = hashlib.sha1(json.dumps(record)).hexdigest()
+    fad_record['size'] = sys.getsizeof(record)
+    fad_record['version'] = sas_version
+    if 'cbsd' in url:
+      fad_record['recordType'] = "cbsd"
+    elif 'zone' in url:
+      fad_record['recordType'] = "zone"
+    elif 'esc_sensor' in url:
+      fad_record['recordType'] = "esc_sensor"
+    else:
+      raise Exception("ConfigurationError:: Incorrect Record Type received in URL:{}".format(url))
+    return fad_record
+
+  @staticmethod
+  def createFadObject(all_activity_dump):
+    """Creates the FAD object format and embed the active dump file object of different recordtype.
+    """
+    full_activity_dump = {}
+    full_activity_dump['files'] = all_activity_dump
+    end_time = datetime.now()
+    end_time = time.strftime(end_time.replace(microsecond=0).isoformat()) + 'Z'
+    full_activity_dump['generationDateTime'] = end_time
+    full_activity_dump['description'] = 'Load FAD to SAS UUT'
+    return full_activity_dump
+
+  @staticmethod
+  def generateCbsdRecords(cbsd_records, grant_records_list):
```

Comment:
> We have redesigned the SAS test harness to address this comment. The configuration file now contains the cabs records, ppa record and esc record. Utility functions are used to create cabs, ppa and esc records conforming to the SAS-SAS interface format from the  existing record types in the testdata folder. The related functions are written as static methods so that they can be invoked within the generate default config method before creating an object for SAS test harness.

This should not be in this class; putting the code here will create confusion. Please move outside the class (in this file is OK) and then I will review these pieces. Right now it's hard to tell what is an integral part of the SAS TH vs. what is just there to help test dev.

Reply:
Ok. We   agree. We have moved the helper methods    as non class functions at the end of the file.